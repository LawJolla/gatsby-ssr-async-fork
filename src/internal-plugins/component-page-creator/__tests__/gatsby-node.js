const validatePath = require(`../validate-path`)
const createPath = require(`../create-path`)

describe(`JavaScript page creator`, () => {
  it(`filters out files that start with underscores`, () => {
    const files = [
      {
        path: `something/blah.js`,
      },
      {
        path: `something/_blah.js`,
      },
      {
        path: `_blah.js`,
      },
    ]

    expect(files.filter(file => validatePath(file.path)).length).toEqual(1)
  })

  it(`filters out files that start with template-*`, () => {
    const files = [
      {
        path: `something/blah.js`,
      },
      {
        path: `something/template-cool-page-type.js`,
      },
      {
        path: `template-cool-page-type.js`,
      },
    ]

    expect(files.filter(file => validatePath(file.path)).length).toEqual(1)
  })

  describe(`create-path`, () => {
    it(`should create unix paths`, () => {
      const basePath = `/a/`
      const paths = [`/a/b/c/de`, `/a/bee`, `/a/b/d/c/`]

      expect(paths.map(p => createPath(basePath, p))).toMatchSnapshot()
    })

    it(`should deal with windows paths`, () => {
      const basePath = `D:/Projets/gatsby-starter/src/pages`
      const paths = [
        `D:\\Projets\\gatsby-starter\\src\\pages\\404.tsx`,
        `D:\\Projets\\gatsby-starter\\src\\pages\\index.tsx`,
        `D:\\Projets\\gatsby-starter\\src\\pages\\blog.tsx`,
      ]

      expect(paths.map(p => createPath(basePath, p))).toMatchSnapshot()
    })
  })
})
