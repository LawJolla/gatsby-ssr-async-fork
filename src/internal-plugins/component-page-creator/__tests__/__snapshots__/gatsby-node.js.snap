// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JavaScript page creator create-path should create unix paths 1`] = `
Array [
  "/b/c/de/",
  "/bee/",
  "/b/d/c/",
]
`;

exports[`JavaScript page creator create-path should deal with windows paths 1`] = `
Array [
  "/404/",
  "/",
  "/blog/",
]
`;
